### ISSUE TITLE
Training code 

### ISSUE THREAD

**Issue by shreyasashwath** [cedriclmenard/irislandmarks.pytorch#1]
Hi cedriclmenard,
Could you please provide the training code for this repo that would be very helpful.
Thanks

**Comment by cedriclmenard** [comment]
Hi @shreyasashwath,

You can take a look at the class definition in irislandmarks.py, especially at the predict and the forward methods. Training can be done as with any PyTorch models if you feed the right data. For a simpler implementation of training, I suggest looking into PyTorch-Lightning.

The goal of this repo is to reproduce the same results using the same weights given by the original authors (Google's Mediapipe), this is why no training is implemented here.

**Comment by sainisanjay** [comment]
> Hi @shreyasashwath,
> 
> You can take a look at the class definition in irislandmarks.py, especially at the predict and the forward methods. Training can be done as with any PyTorch models if you feed the right data. For a simpler implementation of training, I suggest looking into PyTorch-Lightning.
> 
> The goal of this repo is to reproduce the same results using the same weights given by the original authors (Google's Mediapipe), this is why no training is implemented here.

@cedriclmenard Yes agree training can be done using any PyTorch model with right data. 
Can we train this model with few key-points like if i have to extend this to face landmarks (98 keypoints)? and which places we should use Batch normalised layer ?

**Comment by cedriclmenard** [comment]
For face landmarks, I suggest looking at the [FaceMesh](https://github.com/thepowerfuldeez/facemesh.pytorch) repo. You can definitely use this model, but you'll have to modify it according to the output size you need.

Batch norm layers, if there were any in the original training implementation by Google Mediapipe, have been folded in the convolution weights, so you can figure where batch norm layers could go given that.
